### ISSUE TITLE
[MainUI] Option to create Items when adding a thing from the inbox

### ISSUE THREAD

**Issue by mueller-ma** [openhab/openhab-webui#2897]
## The problem

Currently it reqiures several clicks from accepting a thing in the inbox until it can be used. By that I mean that channels need to be linked to Items. Personally I always link channels to new Items. This can be cumbersome when adding many things.

## Your suggestion

Add a new option in the inbox ("Add as Thing and link channels") here:

![grafik](https://github.com/user-attachments/assets/28fc96ba-4b80-4736-828c-436ce9e9660b)

## Your environment

openHAB 4.2.3

## Additional information

Two users told me that this is the largest issue they have with openHAB.

**Comment by rkoshak** [comment]
From a usability perspective, we need to make sure we think about:

- accepting multiple things from the Inbox, 
    - does it go through each thing one-by-one presenting the "Add equipment to model" dialog for that Thing? 
    - how to handle creating new Items verses linking to existing Items. I'm not sure you can do both from any of the current linking screens except when linking Channels one-by-one.
    - can I cancel just one Thing? "Oops, I didn't mean to accept that one but I want all the rest."
- make sure we can still change the Thing ID

Personally, I'd be happy if I accepted one Thing that *that* Thing's page came up next instead of the main Things page with the new Thing burried in the list. If I just accepted a Thing, I'm going to want to work with that Thing. If multiple Things were added in one go, show those Things at the top, or have a "sort by added time" option on the Things page.

**Comment by mueller-ma** [comment]
> does it go through each thing one-by-one presenting the "Add equipment to model" dialog for that Thing?

Yes, but there should also be a "skip" button if one doesn't use the model at all.

> how to handle creating new Items verses linking to existing Items. I'm not sure you can do both from any of the current linking screens except when linking Channels one-by-one.

Form my personal usage I guess that creating new Items is whats usually wanted, but that might be wrong.

> can I cancel just one Thing? "Oops, I didn't mean to accept that one but I want all the rest."

At that stage I wouldn't add a cancel option. You check the thing you want to add before, so there you can skip one. After adding a wrong one, you can remove it quite easy again.

> make sure we can still change the Thing ID

Is this something that's actually required? I never changed the ID of a thing.

**Comment by rkoshak** [comment]
> Form my personal usage I guess that creating new Items is whats usually wanted, but that might be wrong.

During migrations and upgrades between major versions people might have imported a .items file and want to link the Channels to those. 

I too mainly just create new Items but that's because linking Channels to existing Items is so much more painful than deleting the Items and recreating them to reestablish the links. If I had a page simlar to add equipment to model where I can choose the Items instead of creating a new Item that would greatly improve that process.

> Is this something that's actually required? I never changed the ID of a thing.

I *always* change the Thing ID before creating it. Like Item names, Thing IDs  (all IDs really, to include rules, transformations, et al) should, IMO, be meaningful. 

It also has benefits. For example, if you have to recreate a Thing like your Zwave Controller Thing, if you can delete the old, add it again but set the Thing ID to be the same as the old one, all your Links will still work. In the case of bindings with bridges, no only will that one Thing continue to work but all the Things under that bridge will also work in those cases where the bridge ID is in the Thing ID.

But that's why that "Add as Thing (with custom ID)" exists as an option. Se we have to option to set it to something of our choosing before the Thing is created since it cannot be changed after it's created short of manually editing the JSONDB or recreating it.

**Comment by jimtng** [comment]
I think #3020 would help. It gives you the option to go directly to Thing edit, then you can add the items using the same feature from Thing Edit.
